frontend/server/src/Controllers/Course.php

```diff
@@ -20,6 +20,9 @@
  * @psalm-type CourseListPayload=array{courses: array{admin: array{accessMode: string, activeTab: string, filteredCourses: array{current: array{courses: list<FilteredCourse>, timeType: string}, past: array{courses: list<FilteredCourse>, timeType: string}}}, student: array{accessMode: string, activeTab: string, filteredCourses: array{current: array{courses: list<FilteredCourse>, timeType: string}, past: array{courses: list<FilteredCourse>, timeType: string}}}, public: array{accessMode: string, activeTab: string, filteredCourses: array{current: array{courses: list<FilteredCourse>, timeType: string}, past: array{courses: list<FilteredCourse>, timeType: string}}}}}
  * @psalm-type CourseProblemTried=array{alias: string, title: string, username: string}
  * @psalm-type CourseSubmissionsListPayload=array{solvedProblems: array<string, list<CourseProblemTried>>, unsolvedProblems: array<string, list<CourseProblemTried>>}
+ * @psalm-type CourseStudent=array{name: null|string, progress: array<string, float>, username: string}
+ * @psalm-type StudentsProgressPayload=array{course: CourseDetails, students: CourseStudent[], student?: string}
```

Comment:
```suggestion
 * @psalm-type StudentsProgressPayload=array{course: CourseDetails, students: list<CourseStudent>, student?: string}
```

con eso ya no es necesario todo el awkwardness de `Object.values(...)` en el typescript.

Reply:
Done